Userland/Libraries/LibGfx/ImageFormats/PNGWriter.cpp

```diff
@@ -183,23 +183,25 @@ ErrorOr<void> PNGWriter::add_IDAT_chunk(Gfx::Bitmap const& bitmap, Compress::Zli
         struct Filter {
             PNG::FilterType type;
             ByteBuffer buffer {};
-            int sum = 0;
+            AK::SIMD::i32x4 sum { 0, 0, 0, 0 };
 
-            ErrorOr<void> append(u8 byte)
+            ErrorOr<void> append(AK::SIMD::u8x4 simd)
             {
-                TRY(buffer.try_append(byte));
-                sum += static_cast<i8>(byte);
+                TRY(buffer.try_append(simd[0]));
+                TRY(buffer.try_append(simd[1]));
+                TRY(buffer.try_append(simd[2]));
+                if constexpr (include_alpha)
+                    TRY(buffer.try_append(simd[3]));
```

Comment:
Getting this into one `try_append` is likely also nicer

Reply:
I repushed this with slightly more granular commits that maybe make this clearer.

The change is that before we did `sum += (i8)byte;` 3 or 4 times for each pixel's channel, but now we do a single `sum_vec_i32 += (vec_i8)pixel` per pixel (…and then sum up the four channel counts once at the very end).

So it replaces 3 or 4 additions per pixel with a single addition.

Reply:
I bet there's a ton more we can do here :)

The actual deflate code can also use some more speed as well.